### PR TITLE
[package:http_profile] Expand README.md

### PR DESCRIPTION
**Pull request by derekxu16**

### CODE REVIEW

File: pkgs/http_profile/README.md [hunk 1]
```diff
@@ -1,2 +1,68 @@
-An **experimental** package that allows HTTP clients outside of the Dart SDK
-to integrate with the DevTools Network tab.
+[![pub package](https://img.shields.io/pub/v/http_profile.svg)](https://pub.dev/packages/http_profile)
+[![package publisher](https://img.shields.io/pub/publisher/http_profile.svg)](https://pub.dev/packages/http_profile/publisher)
+
+A package that allows HTTP clients outside of the Dart SDK to integrate with
+the DevTools Network tab.
+
```
Comment (brianquinlan) [inline pkgs/http_profile/README.md:6]: Maybe:

**NOTE:** This package is mean for developers *implementing* HTTP clients, not developers *using* HTTP clients.

File: pkgs/http_profile/README.md [hunk 2]
```diff
@@ -1,2 +1,68 @@
...
+## Using
+
+`HttpClientRequestProfile.profile` returns an `HttpClientRequestProfile` object.
```
Comment (brianquinlan) [inline pkgs/http_profile/README.md:9]: ...object if HTTP profiling is enabled.

File: pkgs/http_profile/README.md [hunk 3]
```diff
@@ -1,2 +1,68 @@
...
+Populating the fields of that object with information about an HTTP request and
+about the response to that request will make that information show up in the
+DevTools Network tab.
```
Comment (brianquinlan) [inline pkgs/http_profile/README.md:12]: [DevTools Network View](https://docs.flutter.dev/tools/devtools/network).

File: pkgs/http_profile/README.md [hunk 4]
```diff
@@ -1,2 +1,68 @@
...
+
+```dart
+import 'package:http_profile/http_profile.dart';
+
+void main() {
+  HttpClientRequestProfile.profilingEnabled = true;
```
Comment (brianquinlan) [inline pkgs/http_profile/README.md:18]: I'd remove this because it is not idiomatic for real code.

File: pkgs/http_profile/README.md [hunk 5]
```diff
@@ -1,2 +1,68 @@
...
+
+  final profile = HttpClientRequestProfile.profile(
+    requestStartTime: DateTime.parse('2024-03-21'),
+    requestMethod: 'GET',
+    requestUri: 'https://www.example.com',
+  )!
```
Comment (brianquinlan) [inline pkgs/http_profile/README.md:24]: I'd remove the `!` and use the `?..` cascade.

File: pkgs/http_profile/README.md [hunk 6]
```diff
@@ -1,2 +1,68 @@
...
+
+`HttpClientRequestProfile.profile` returns an `HttpClientRequestProfile` object.
+Populating the fields of that object with information about an HTTP request and
+about the response to that request will make that information show up in the
+DevTools Network tab.
+
+```dart
+import 'package:http_profile/http_profile.dart';
+
+void main() {
```
Comment (brianquinlan) [inline pkgs/http_profile/README.md:17]: Maybe make this look more realistic by calling this function something like:

```dart
Future<String> get(Uri uri) {
  final profile = HttpClientRequestProfile.profile(
    requestStartTime: DateTime.now(),
    requestMethod: 'GET',
    requestUri: uri.toString(),
  );
  profile?.connectionInfo = {
      'localPort': 1285,
      'remotePort': 443,
      'connectionPoolId': '21x23',
    };

  profile.requestData.proxyDetails = HttpProfileProxyData(
    host: 'https://www.example.com',
    username: 'abc123',
    isDirect: true,
    port: 4321,
  );

  // Make the HTTP request and populate the response data.

  profile.requestData.proxyDetails = HttpProfileProxyData(
    host: 'https://www.example.com',
    username: 'abc123',
    isDirect: true,
    port: 4321,
  );
  profile.responseData.headersListValues = {
    'connection': ['keep-alive'],
    'cache-control': ['max-age=43200'],
    'content-type': ['application/json', 'charset=utf-8'],
  };

  return responseString;
}
```
  ...

}```

File: pkgs/http_profile/README.md [hunk 7]
```diff
@@ -1,2 +1,68 @@
...
+    'cache-control': ['max-age=43200'],
+    'content-type': ['application/json', 'charset=utf-8'],
+  };
+
+  // ...
+
+}
+```
+
+Refer to the source of
```
Comment (brianquinlan) [inline pkgs/http_profile/README.md:49]: That's a great idea!

File: pkgs/http_profile/README.md (hunk 1 above)
Comment (derekxu16) [reply pkgs/http_profile/README.md:6]: Done

File: pkgs/http_profile/README.md (hunk 2 above)
Comment (derekxu16) [reply pkgs/http_profile/README.md]: Done.

File: pkgs/http_profile/README.md (hunk 3 above)
Comment (derekxu16) [reply pkgs/http_profile/README.md]: Done. Should I also change "Network tab" to "Network View" on line 5?

File: pkgs/http_profile/README.md (hunk 4 above)
Comment (derekxu16) [reply pkgs/http_profile/README.md]: Done

File: pkgs/http_profile/README.md (hunk 6 above)
Comment (derekxu16) [reply pkgs/http_profile/README.md]: Done

File: pkgs/http_profile/README.md (hunk 3 above)
Comment (brianquinlan) [reply pkgs/http_profile/README.md]: Yes please.

**Review by brianquinlan — APPROVED** [review]

File: pkgs/http_profile/README.md (hunk 3 above)
Comment (derekxu16) [reply pkgs/http_profile/README.md]: Done

**Pull request merged**